packages/gitbook/src/lib/api.ts

```diff
@@ -722,7 +722,7 @@ export const getSiteSpaces = cache(
             organizationId: string;
             siteId: string;
             /** Site share key that can be used as context to resolve site space published urls */
-            siteShareKey: string | undefined;
+            siteShareKey?: string;
```

Comment:
We should not make it optional. It causes bugs. It was typed like this so we always specify a `shareKey` when we have it otherwise the urls in the object is not properly resolved (for share link sites)

Reply:
Why are typing it as potentially undefined in that case?

Reply:
my guess is that what we really want here then is `string | null`

Reply:
Yes `string | null` is fine but we absolutely need to not make it optional